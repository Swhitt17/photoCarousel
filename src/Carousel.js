import React, {useState} from "react";
import Card from "./Card"
import "./Carousel.css"


const Carousel = ({photos, title}) => {
    const [currCardInd, setCurrCardInd] = useState(0);
 

    const currCard = photos[currCardInd];
    const total = photos.length;

    function goForward(){
        setCurrCardInd(currCardInd + 1);
    }

    function goBackward(){
        setCurrCardInd(currCardInd - 1);
    }


    return (
        <div className="Carousel">
            <h1>{title}</h1>
            <div className="Carousel=main"> 
            
            <i
            className={currCardInd === 0 ? "bi bi-hidden-left-arrow-left-circle" : "bi bi-arrow-left-circle"}
            onClick={goBackward}
            />
            <Card 
             caption={currCard.caption}
             src={currCard.src}
             currNum={currCardInd + 1}
             totalNum={total}
            />
            
            <i 
            className={currCardInd === 2 ? "bi bi-hidden-right-arrow-right-circle" : "bi bi-arrow-right-circle"}
            onClick={goForward}
            />
           
            </div>
        </div>
    );
}

export default Carousel;