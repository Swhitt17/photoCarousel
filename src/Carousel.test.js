import React from "react";
import {render, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom';
import Carousel from "./Carousel";
import TEST_IMAGES from "./_testCommon";


test("it renders without crashing", () => {
    
    render(
    <Carousel photos={TEST_IMAGES} title="test images"/>);
})

test("it matches snapshot", () => {
    const {asFragment} = render(
    <Carousel photos={TEST_IMAGES} title="test images"/>);
    expect(asFragment()).toMatchSnapshot()
})


test("left arrow goes back an image", () => {
    const {container} = render(
        <Carousel photos={TEST_IMAGES} title="test images"/>)
    const leftArrow = container.querySelector(".bi-arrow-left-circle");
    const secondImage = container.querySelector('img[caption="testing image 2"')
    const firstImage = container.querySelector('img[caption="testing image 1"')
    expect(secondImage).toBe(secondImage)
    fireEvent.click(leftArrow);
    expect(firstImage).toBe(firstImage)
})


test("left arrow is removed on first image", () => {
    const {container} = render(
        <Carousel photos={TEST_IMAGES} title="test images"/>)
    const leftArrow = container.querySelector(".bi-arrow-left-circle");
    const secondImage = container.querySelector('img[src="test2.com"')
    const firstImage = container.querySelector('img[src="test1.com"')
    expect(secondImage).toBe(secondImage)
    expect(leftArrow).toBeInTheDocument
    fireEvent.click(leftArrow);
    expect(firstImage).toBe(firstImage)
    expect(leftArrow).not.toBeInTheDocument
})

test("right arrow is removed on last image", () => {
    const {container} = render(
        <Carousel photos={TEST_IMAGES} title="test images"/>)
    const rightArrow = container.querySelector(".bi-arrow-right-circle");
    const secondImage = container.querySelector('img[src="test2.com"')
    const thirdImage = container.querySelector('img[src="test3.com"')
    expect(secondImage).toBe(secondImage)
    expect(rightArrow).toBeInTheDocument
    fireEvent.click(rightArrow);
    expect(thirdImage).toBe(thirdImage)
    expect(rightArrow).toBeInTheDocument
})
