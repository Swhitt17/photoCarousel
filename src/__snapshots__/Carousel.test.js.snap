// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Carousel"
  >
    <h1>
      test images
    </h1>
    <div
      class="Carousel=main"
    >
      <i
        class="bi bi-hidden-left-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        />
        <img
          class="Card-image"
        />
        <small
          class="Card-small"
        >
          Image 1 of 11
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</DocumentFragment>
`;
